### ISSUE TITLE
GDumb strategy performance issues with Rotated/Permuted MNIST benchmarks

### ISSUE THREAD

**Issue by Ektagavas** [ContinualAI/avalanche#765]
When running GDumb for MNIST variants where the number of classes remains the same (from 0 to 9) in each experience,  accuracy on all old tasks drops significantly with the newer task. I wanted to confirm this is a bug in the Gdumb strategy in avalanche or I am missing something on my part.

Below is the script to reproduce the results for Rotated MNIST:
```
train_transform = Compose([
        ToTensor(),
        Normalize((0.1307,), (0.3081,))
    ])
eval_transform = Compose([
        ToTensor(),
        Normalize((0.1307,), (0.3081,))
    ])
scenario = RotatedMNIST(
    n_experiences=2, 
    train_transform=train_transform, 
    eval_transform=eval_transform,
    seed=1234)

interactive_logger = InteractiveLogger()

eval_plugin = EvaluationPlugin(
    accuracy_metrics(experience=True, stream=True),
    loss_metrics(experience=True, stream=True),
    loggers=[interactive_logger])

model = SimpleMLP(hidden_size=100, hidden_layers=2, num_classes=10, drop_rate=0.)
optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
criterion = torch.nn.CrossEntropyLoss()

strategy = Naive(
            model, optimizer, criterion,
            train_mb_size=16, device=device,
            eval_mb_size=16, train_epochs=15,
            plugins=[GDumbPlugin(mem_size=500)]
        )

print('Starting experiment...')
for experience in scenario.train_stream:
  print("Start training on experience ", experience.current_experience)

  strategy.train(experience)
  print("End training on experience ", experience.current_experience)
  print('Computing accuracy on the test set')
  strategy.eval(scenario.test_stream[:])
```

Below are the results:
After training on Exp0:
Top1_Acc_Stream/eval_phase/test_stream/Task000 = 0.8488
Top1_Acc_Stream/eval_phase/test_stream/Task001 = 0.1460

After training on Exp1:
Top1_Acc_Stream/eval_phase/test_stream/Task000 = 0.1373
Top1_Acc_Stream/eval_phase/test_stream/Task001 = 0.7879

Similar trend is observed for PermutedMNIST as well.

Thank you!

**Comment by Ektagavas** [comment]
What I understood from GDumb paper is that, during inference, it masks out the subset of stored examples depending on task and performs classification. I guess this step might be missing in avalanche as the storage policy just samples, stores and concats examples to set based on class labels (which is 0 to 9 for each experience). Please correct me if I am wrong.

**Comment by AntonioCarta** [comment]
The masking is basically the same thing as a multihead (in evaluation mode). I believe the storage policy may be wrong when the same class occurs in multiple experiences, making the buffers unbalanced. I will check this and let you know.

**Comment by Ektagavas** [comment]
> The masking is basically the same thing as a multihead (in evaluation mode). I believe the storage policy may be wrong when the same class occurs in multiple experiences, making the buffers unbalanced. I will check this and let you know.

Yes, correct. Sure, thanks.
